system/Commands/Utilities/Optimize.php

```diff
@@ -99,32 +115,73 @@ private function removeFile(string $cache): void
-            $config,
-            [
-                'public bool $configCacheEnabled = false;'  => 'public bool $configCacheEnabled = true;',
-                'public bool $locatorCacheEnabled = false;' => 'public bool $locatorCacheEnabled = true;',
-            ]
-        );
+        // Prepare search and replace mappings
+        $searchReplace = [];
+
+        if ($disable === true) {
+            $searchReplace = [
+                'public bool $configCacheEnabled = true;'  => 'public bool $configCacheEnabled = false;',
+                'public bool $locatorCacheEnabled = true;' => 'public bool $locatorCacheEnabled = false;',
+            ];
+        } else {
+            if ($enableConfigCache === true) {
+                $searchReplace['public bool $configCacheEnabled = false;'] = 'public bool $configCacheEnabled = true;';
+            }
 
-        if ($result) {
-            CLI::write(
-                'Config Caching and FileLocator Caching are enabled in "app/Config/Optimize.php".',
-                'green'
-            );
+            if ($enableLocatorCache === true) {
+                $searchReplace['public bool $locatorCacheEnabled = false;'] = 'public bool $locatorCacheEnabled = true;';
+            }
 
-            return;
+            // If no options provided, update both
+            if ($enableConfigCache === null && $enableLocatorCache === null) {
+                $searchReplace = [
+                    'public bool $configCacheEnabled = false;'  => 'public bool $configCacheEnabled = true;',
+                    'public bool $locatorCacheEnabled = false;' => 'public bool $locatorCacheEnabled = true;',
+                ];
+            }
         }
 
-        CLI::error('Error in updating file: ' . clean_path($config));
+        // Apply replacements if necessary
+        if ($searchReplace !== []) {
+            $result = $publisher->replace($config, $searchReplace);
 
-        throw new RuntimeException(__METHOD__);
+            if ($result === true) {
+                $messages = [];
+
+                if (in_array('public bool $configCacheEnabled = true;', $searchReplace, true)) {
+                    $messages[] = 'Config Caching is enabled in "app/Config/Optimize.php".';
+                }
+
+                if (in_array('public bool $locatorCacheEnabled = true;', $searchReplace, true)) {
+                    $messages[] = 'FileLocator Caching is enabled in "app/Config/Optimize.php".';
+                }
+
+                if (in_array('public bool $configCacheEnabled = false;', $searchReplace, true)) {
+                    $messages[] = 'Config Caching is disabled in "app/Config/Optimize.php".';
+                }
+
+                if (in_array('public bool $locatorCacheEnabled = false;', $searchReplace, true)) {
+                    $messages[] = 'FileLocator Caching is disabled in "app/Config/Optimize.php".';
+                }
+
+                CLI::write(implode("\n\n", $messages), 'green');
```

Comment:
Why do we need two `\n`?
And does `\n` work on Windows?

Reply:
It's used to create a new line to format the message more nicely. I'm not sure if it works on all platform.

I tried asking GPT and the matrix given as follow:

"Here is the matrix showing support for using printf "\n" to print a new line across various platforms and shells:"
Platform | Bash | Zsh | Fish | Ksh | Cmd | Powershell
-- | -- | -- | -- | -- | -- | --
Linux | Yes | Yes | Yes | Yes | No | No
macOS | Yes | Yes | Yes | Yes | No | No
Windows (WSL, Git Bash) | Yes | Yes | Yes | Yes | No | No
Windows (cmd) | No | No | No | No | Yes (with workaround) | Yes (with workaround)

Reply:
If you did not check on Windows, please use `PHP_EOL` instead of `\n`.

Reply:
i will replace that with `CLI::newLine();`